### PR TITLE
MM-41323: Secure Marketplace URL

### PR DESCRIPTION
**Pull request by agnivade**
Disable marketplace URL text box when
plugin uploads are disabled.

https://mattermost.atlassian.net/browse/MM-41323

### CODE REVIEW

File: e2e/cypress/tests/integration/system_console/plugin_marketplace_url_spec.js [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+            PluginSettings: {
+                Enable: true,
+            },
+        };
+        cy.apiUpdateConfig(newSettings);
+
+        cy.exec('tests/fixtures/mmctl --local config set PluginSettings.EnableUploads false');
+
+        // # Login as System Admin
+        cy.apiAdminLogin();
```
Comment (saturninoabril) [inline e2e/cypress/tests/integration/system_console/plugin_marketplace_url_spec.js:32]: Can be removed as well.

**Review by saturninoabril — CHANGES_REQUESTED** [review]
Thanks @agnivade! LGTM, except for few comment in test. Please fix the build then I'll run the test in Gitlab with mmctl.

**Review by srkgupta — APPROVED** [review]
Thanks @agnivade. Changes LGTM

**Review by saturninoabril — APPROVED** [review]
Thanks @agnivade! Tested and passed. Looks good to me. For test infra, I'll submit an update to share volume of socket path/file to Cypress container, so that `--local` will work.

**Pull request merged**